src/main/java/org/spongepowered/api/event/entity/living/player/PlayerResourcePackStatusEvent.java

```diff
@@ -0,0 +1,63 @@
+package org.spongepowered.api.event.entity.living.player;
+
+import org.spongepowered.api.resourcepack.ResourcePack;
+
+import com.google.common.base.Optional;
+
+public interface PlayerResourcePackStatusEvent extends PlayerEvent {
+
+    /**
+     * Gets the pack that this status corresponds to.
+     * 
+     * @return The pack that this status corresponds to.
+     */
+    ResourcePack getPack();
+
+    /**
+     * Gets the status of the sent pack.
+     * 
+     * @return The status of the sent pack.
+     */
+    ResourcePackStatus getStatus();
+
+    public enum ResourcePackStatus {
+        /**
+         * The client is attempting to download the pack.
+         */
+        ACCEPTED(Optional.<Boolean> absent()),
```

Comment:
It's an enum, not a quasi-enum, without reflection you can't create a new instance. There's no case of extensibility here, either, so I don't see the issue.